tests/scenarios.go

```diff
@@ -431,11 +439,37 @@ func addParameterFrom(t gobdd.StepTest, ctx gobdd.Context, name string, path str
 	ctx.Set(requestArgsKey{}, append(GetRequestArguments(ctx), value))
 }
 
+// Given a value, coerce the proper type that the request expects, and add the path param to the context
+// This shouldn't be used as a "step" method, and is a helper utility
+func addPathArgumentWithValue(t gobdd.StepTest, ctx gobdd.Context, param string, value string, request reflect.Value) {
+	in := make([]reflect.Value, request.Type().NumIn())
+	in[0] = reflect.ValueOf(GetCtx(ctx))
+	// The order of the path arguments in the scenario definition
+	// must match the order of the arguments in the function signature
+	// Here we keep track of which numbered path argument we're setting
+	pathCount, _ := ctx.Get(pathParamCountKey{})
+	varType := reflect.New(request.Type().In(pathCount.(int)))
+	ctx.Set(pathParamCountKey{}, pathCount.(int)+1)
+
+	templatedValue := Templated(t, GetData(ctx), value)
+
+	json.Unmarshal([]byte(templatedValue), varType.Interface())
+	GetRequestParameters(ctx)[param] = varType.Elem()
+
+	ctx.Set(requestArgsKey{}, append(GetRequestArguments(ctx), varType.Elem()))
+}
+
 func addParameterWithValue(t gobdd.StepTest, ctx gobdd.Context, param string, value string) {
 	// Get request builder for the current scenario
 	request, err := getRequestBuilder(ctx)
+
+	// If the getRequestBuilder method returns a Function, its likely because we're trying to add a
+	// path param instead of a query param.
 	if err != nil {
-		t.Error(err)
+		if request.Type().Kind() == reflect.Func {
+			addPathArgumentWithValue(t, ctx, param, value, request)
+			return
+		}
```

Comment:
`err` would be nil here, so this should be out of the if. `getRequestBuilder` can still return an error, so we still want to fail the test in that case